packages/@lwc/engine/src/framework/def.ts

```diff
@@ -145,9 +146,15 @@ function createComponentDef(
         renderedCallback = renderedCallback || superDef.renderedCallback;
         errorCallback = errorCallback || superDef.errorCallback;
         render = render || superDef.render;
+        template = template || superDef.template;
```

Comment:
this is the meat of the PR, we use the template definition from the super if the subclass doesn't have one.